tests/protocols/test_websocket.py

```diff
@@ -466,3 +466,32 @@ async def get_subprotocol(url):
         accepted_subprotocol = loop.run_until_complete(get_subprotocol(url))
         assert accepted_subprotocol == subprotocol
         loop.close()
+
+
+@pytest.mark.parametrize("protocol_cls", WS_PROTOCOLS)
+def test_server_lost_connection(protocol_cls, reraise):
+    class App(WebSocketResponse):
+        async def websocket_connect(self, message):
+            await self.send({"type": "websocket.accept"})
+            await self.receive()
+
+            # Simulate a lost connection
+            # without receiving a close frame
+            self.send.__self__.connection_lost(None)
+
+            with reraise:
+                await self.send({"type": "websocket.send", "text": "123"})
+
+    async def websocket_session(url):
+        async with websockets.connect(url) as websocket:
+            await websocket.ping()
+            await websocket.send("abc")
+            # Delay exiting context manager
+            # to avoid sending a close frame before server attempts send
+            await asyncio.sleep(1)
+
+    if protocol_cls is WebSocketProtocol:
```

Comment:
I would let this run on both implementations, neither should raise exceptions.

Reply:
yep, didn't see that >_ fixed